src/postgres_utils.py

```diff
@@ -0,0 +1,167 @@
+        """,
+        {"table_schema": table_schema},
+    )
+
+    return sorted(table_names["table_name"].to_list())
+
+
+def get_table_columns(table_schema: str, table_name: str) -> list:
+    column_names = execute_sql_select_query(
+        """
+        SELECT column_name FROM information_schema.columns
+        WHERE table_schema = ':table_schema'
+        AND table_name   = ':table_name';
+        """,
+        {"table_schema": AsIs(table_schema), "table_name": AsIs(table_name)},
+    )
+    return sorted(column_names["column_name"])
+
+
+def get_table_row_count(table_schema: str, table_name: str) -> int:
+    row_counts = execute_sql_select_query(
+        """
+        SELECT c.reltuples::bigint AS row_count
+        FROM pg_class c
+        JOIN pg_namespace n ON n.oid = c.relnamespace
+        WHERE c.relname = ':table_name'
+        AND n.nspname = ':table_schema';
+        """,
+        {"table_schema": AsIs(table_schema), "table_name": AsIs(table_name)},
+    )
+    return int(row_counts["row_count"][0])
+
+
+def create_postigs_extension() -> None:
+    query = "CREATE EXTENSION POSTGIS"
+    execute_sql_query(query)
+
+
+def vacuum_database() -> None:
+    query = "VACUUM (ANALYZE)"
+    execute_sql_query(query)
+
+
+def create_sql_schema(table_schema: str) -> pd.DataFrame:
+    execute_sql_query(
+        "DROP SCHEMA IF EXISTS :table_schema CASCADE",
+        {"table_schema": AsIs(table_schema)},
+    )
+    execute_sql_query(
+        "CREATE SCHEMA :table_schema", {"table_schema": AsIs(table_schema)}
+    )
+    table_schema_name = execute_sql_select_query(
+        """
+        SELECT schema_name
+        FROM :table_schema.schemata;
+        """,
+        {"table_schema": AsIs("information_schema")},
+    )
+    return table_schema_name
+
+
+def execute_sql_select_query(query: str, placeholders: dict = {}) -> pd.DataFrame:
+    sql_engine = create_engine(BUILD_ENGINE)
+    with sql_engine.begin() as sql_conn:
+        select_records = pd.read_sql(sql=text(query), con=sql_conn, params=placeholders)
+    sql_engine.dispose()
+    return select_records
+
+
+def execute_sql_query(query: str, placeholders: dict = {}) -> None:
+    sql_engine = create_engine(BUILD_ENGINE, isolation_level="AUTOCOMMIT")
+    with Session(sql_engine) as session:
+        session.execute(statement=text(query), params=placeholders)
+        session.commit()
+    sql_engine.dispose()
+
+
+def execute_sql_file(filename: str) -> None:
+    print(f"Executing {SQL_FILE_DIRECTORY / filename} ...")
+    sql_file = open(SQL_FILE_DIRECTORY / filename, "r")
+    sql_query = ""
+    for line in sql_file:
+        # ignore comment lines
+        if line.strip("\n") and not line.startswith("--"):
+            # append line to the query string
+            sql_query += line.strip("\n")
+            # if the query string ends with ';', it is a full statement
+            if sql_query.endswith(";"):
+                execute_sql_query(query=sql_query)
```

Comment:
Just making sure this is desired behavior - this will commit each separate statement within the file, so if it fails at some point, some fraction of the file will have been successfully run. If we'd rather have all or nothing, could simply do something like 

```python
with engine.connect() as con:
    with open("src/models/query.sql") as file:
        query = text(file.read())
        con.execute(query)
```

Reply:
good call, but may be irrelevant per [comment above](https://github.com/NYCPlanning/data-engineering-qaqc/pull/258#discussion_r1170571660)